### ISSUE TITLE
Push mode with dock pushes content under sidebar in IE 11 

### ISSUE THREAD

**Issue by Akkusativobjekt** [arkon/ng-sidebar#104]
When the sidebar is in push mode and dock is set to true, the sidebar does not behave as expected in IE 11. I don't know if this affects other IE versions, because I have only IE 11 installed.

As the gif shows the content is pushed under the sidebar.
![test](https://user-images.githubusercontent.com/5674024/27730778-d57bc446-5d8a-11e7-9042-5792cd267c51.gif)

The used version of ng-sidebar is the newest one (6.0.1). 

Also keep up the good work, I really appreciate what you are doing :wink:

**Comment by arkon** [comment]
Geh, I think this is because of the fix I introduced for #103. I'm assuming it happens for _all_ modes when docked?

**Comment by Akkusativobjekt** [comment]
I also believe that the last fix introduced this bug, too.

I made a fast (additional) test with the IE 11:
dock + over => Does not work correctly, the content is now below the sidebar
dock + push => Works incorrect like shown in the gif
dock + slide => Does not work correctly, the content is also below the sidebar

So yes you are right, all three modes does not work correctly.

**Comment by arkon** [comment]
@Akkusativobjekt Thanks for testing!

**Comment by TaylorZaneKirk** [comment]
This is an issue I just stumbled upon as well. Hope for a fix or work-around soon!

**Comment by TaylorZaneKirk** [comment]
I fixed it. According to what I could did up, this is the line causing issues:
```
translateAmt = `calc(${translateAmt} ${this._isLeftOrTop ? '+' : '-'} ${this.dockedSize})`;
```
From what I read IE 11 has a bug that causes "calc()" not to be supported. The worker around I came up with is that "translateX(calc(100% - 100px))" is the same thing as "translateX(100%) translateX(-100px)"

so here's my fix:
```
 _getStyle(): CSSStyleDeclaration {
    let transformStyle: string = null;

    // Hides sidebar off screen when closed
    if (!this.opened) {
      const transformDir: string = this._isLeftOrRight ? 'X' : 'Y';
      let translateAmt: string = `${this._isLeftOrTop ? '-' : ''}100%`;
      let translateDock: string = "";

      // Docked mode: partially remains open
      if (this.dock && this._dockedSize > 0 && !(this._isModeSlide && this.opened)) {
        translateDock = `${this._isLeftOrTop ? '+' : '-'}${this.dockedSize}`;
      }
      else{
        translateDock = "0px";
      }

      transformStyle = `translate${transformDir}(${translateAmt}) translate${transformDir}(${translateDock})`
    }

    return {
      webkitTransform: transformStyle,
      transform: transformStyle
    } as CSSStyleDeclaration;
  }
```

That should be correct. I actually wrote the fix in the JS files from the generated lib directory in my node_modules directory. In case anyone is interested in that:

```
Sidebar.prototype._getStyle = function () {
        var transformStyle = null;
        // Hides sidebar off screen when closed
        if (!this.opened) {
            var transformDir = this._isLeftOrRight ? 'X' : 'Y';
            var translateAmt = (this._isLeftOrTop ? '-' : '') + "100%";
            var translateDock = "";
            // Docked mode: partially remains open
            if (this.dock && this._dockedSize > 0 && !(this._isModeSlide && this.opened)) {
                translateDock= "" + (this._isLeftOrTop ? '+' : '-') + this.dockedSize;

            }
            else{
                translateDock = "0px";
            }
            
            transformStyle= "translate" + transformDir + "(" + translateAmt + ") translate" + transformDir + "(" + translateDock + ")";
            console.log(transformStyle);
        }
        return {
            webkitTransform: transformStyle,
            transform: transformStyle
        };
    };
```

Anyway, the sidebar works perfectly for me now, so I hope this helps someone else.

**Comment by arkon** [comment]
@TaylorZaneKirk Nice work! Sorry I never got around to fixing this. I'll take a look at this tonight and push it if it's all good.

**Comment by arkon** [comment]
Fixed in v6.0.2. Thansk @TaylorZaneKirk !

**Comment by ingarpo** [comment]
After a bunch of hours! Thanks @TaylorZaneKirk  @arkon
